torchmetrics/functional/audio/pesq.py

```diff
@@ -0,0 +1,70 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+import pesq as pesq_backend
+import torch
+from torch import Tensor
+
+from torchmetrics.utilities.checks import _check_same_shape
+
+
+def pesq(preds: Tensor, target: Tensor, fs: int, mode: str, keep_same_device: bool = False) -> Tensor:
+    r"""PESQ (Perceptual Evaluation of Speech Quality)
+
+    This is a wrapper for the pesq package [1].
+
+    Args:
+        preds:
+            shape ``[...,time]``
+        target:
+            shape ``[...,time]``
+        fs:
+            sampling frequency, should be 16000 or 8000 (Hz)
+        mode:
+            'wb' (wide-band) or 'nb' (narrow-band)
+        keep_same_device:
+            whether to move the pesq value to the device of preds
```

Comment:
why is this argument needed? I would expect that we always wanted to move to the same device

Reply:
because this metric can only be computed on cpu, but the preds or target can be on gpu

Reply:
this metric is usually used in testing or validation steps, there is no need to move them to gpu if preds is on gpu, which will slow down the speed of testing or validation